client/components/user-profile.js

```diff
@@ -0,0 +1,307 @@
+            <div style={{ marginTop: "20px" }}>
+              <Button animated="vertical" style={{ fontSize: "1rem" }}>
+                <Button.Content visible>Follow</Button.Content>
+                <Button.Content hidden>
+                  <Icon name="user" color="blue" />
+                </Button.Content>
+              </Button>
+
+              <Button animated="vertical" style={{ fontSize: "1rem" }}>
+                <Button.Content visible>Share</Button.Content>
+                <Button.Content hidden>
+                  <Icon name="share" color="red" />
+                </Button.Content>
+              </Button>
+            </div>
+          </Card.Content>
+          <Form>
+            <TextArea
+              placeholder="Enter Bio"
+              onKeyDown={(e) => handleKeySubmitBio(e, e.target.value)}
+              style={{ height: "8vh" }}
+            />
+          </Form>
+          <Card.Description
+            style={{
+              height: "5vw",
+              width: "30vw",
+              borderStyle: "solid",
+              borderColor: "black",
+              borderWidth: "0.1px",
+              margin: "20px",
+            }}
+          >
+            {currentBio}
+          </Card.Description>
+          <Input
+            placeholder="Guilty Pleasure"
+            onKeyDown={(e) => handleKeySubmitGP(e, e.target.value)}
+          />
+          <Card.Description
+            style={{
+              height: "3vw",
+              width: "30vw",
+              borderStyle: "solid",
+              borderColor: "black",
+              borderWidth: "0.1px",
+              margin: "20px",
+            }}
+          >
+            {guiltyPleasure}
+          </Card.Description>
+        </Card>
+        <Card
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+            width: "40vw",
+            backgroundColor: "rgba(255,255,255,0.9)",
+          }}
+        >
+          <div style={{ width: "30vw" }}>
+            <div>
+              <p style={{ fontSize: "2.5rem", textAlign: "center" }}>
+                Ways I Improve My Mood{" "}
+                <Icon name="square" basic color="yellow" />
+              </p>
+              <Input
+                placeholder="Add to List"
+                onKeyDown={(e) => handleKeySubmitImproveMood(e, e.target.value)}
+              />{" "}
+            </div>
+
+            <List style={{ fontSize: "2rem" }}>
+              <div
+                style={{
+                  display: "flex",
+                  flexDirection: "column",
+                  flexWrap: "wrap",
+                  alignItems: "start",
+                  gap: "20px",
+                  contain: "content",
+                  overflow: "auto",
+                  height: "20vw",
+                  width: "30vw",
+                }}
+              >
+                {/* {currentMoodImprovement.map((item) => {
```

Comment:
This is where I was trying to map it when i had the initial state as an array first